File: app/src/main/java/com/storehop/app/data/db/relations/ShoppingRow.kt
package com.storehop.app.data.db.relations

import androidx.room.ColumnInfo

data class ShoppingRow(
    @ColumnInfo(name = "id") val itemId: String,
    @ColumnInfo(name = "name") val itemName: String,
    @ColumnInfo(name = "quantity") val quantity: String?,
    @ColumnInfo(name = "notes") val notes: String?,
    @ColumnInfo(name = "isNeeded") val isNeeded: Boolean,
    @ColumnInfo(name = "brand") val brand: String?,
    @ColumnInfo(name = "imageUrl") val imageUrl: String?,
    @ColumnInfo(name = "isPriority") val isPriority: Boolean,
    @ColumnInfo(name = "isStaple") val isStaple: Boolean,
    @ColumnInfo(name = "cat_id") val categoryId: String?,
    @ColumnInfo(name = "cat_name") val categoryName: String?,
    /**
     * Seeded category nameKey (e.g. "cat_produce") for resource-string
     * lookup. Null for user-added categories -- in that case the screen
     * falls back to [categoryName] which is whatever the user typed.
     */
    @ColumnInfo(name = "cat_nameKey") val categoryNameKey: String?,
    @ColumnInfo(name = "cat_icon") val categoryIcon: String?,
    @ColumnInfo(name = "displayOrder") val displayOrder: Int?,
)

File: ios/Storehop/Data/DB/ShoppingRow.swift
import Foundation
import GRDB

/// One row of the Shop-at-Store list, denormalized via INNER/LEFT joins on
/// items, item_store_xref, categories, and store_category_order. The ordering
/// fields (`displayOrder`, `categoryName`) come from the joins, not the items
/// table — see `ShoppingDao.shoppingListForStore` for the shape.
///
/// Mirrors the Android `ShoppingRow` data class. Column names in the SQL
/// `AS` aliases below MUST match `CodingKeys` so `FetchableRecord` decodes
/// the join result correctly.
struct ShoppingRow: FetchableRecord, Decodable, Hashable, Sendable {
    let id: String
    let name: String
    let quantity: String?
    let notes: String?
    let isNeeded: Bool
    let brand: String?
    let imageUrl: String?
    let isPriority: Bool
    let isStaple: Bool
    let categoryId: String?
    let categoryName: String?
    let categoryNameKey: String?
    let categoryIcon: String?
    let displayOrder: Int?

    enum CodingKeys: String, CodingKey {
        case id
        case name
        case quantity
        case notes
        case isNeeded
        case brand
        case imageUrl
        case isPriority
        case isStaple
        case categoryId    = "cat_id"
        case categoryName  = "cat_name"
        case categoryNameKey = "cat_nameKey"
        case categoryIcon  = "cat_icon"
        case displayOrder
    }

    /// Convenience for the UI — `id` is the *item's* id, since each item
    /// shows once per shopping list per store. Lets SwiftUI use this row as
    /// `Identifiable` directly.
    var itemId: String { id }
    var itemName: String { name }
}

extension ShoppingRow: Identifiable {}
